api/v1/qdrantclustersnapshot_types.go

```diff
@@ -56,6 +56,9 @@ type VolumeSnapshotInfo struct {
 	// ReadyToUse indicates if the volume snapshot is ready to use
 	// +optional
 	ReadyToUse bool `json:"readyToUse"`
+	// SnapshotHandle is the identifier of the volume snapshot in the respective cloud provider
+	// +optional
+	SnapshotHandle string `json:"snapshotHandle,omitempty"`
```

Comment:
is this the information we want to push to the cluster-api (by the agent) or is this an indirection of e.g. the size?
How does this differ from the volumeSnapshotName?

Reply:
Yes. This is information we want to push to cluster-api. This comes from VolumeSnapshotContent CR. It is the cloud provider identifier for the snapshot. Cluster API use this to query to cloud provider api.

```yaml
apiVersion: snapshot.storage.k8s.io/v1
kind: VolumeSnapshotContent
metadata:
  creationTimestamp: "2024-09-04T13:42:15Z"
  finalizers:
  - snapshot.storage.kubernetes.io/volumesnapshotcontent-bound-protection
  generation: 1
  name: snapcontent-dfeaf5ce-d193-4ec8-8e87-f5db2cf5464a
  resourceVersion: "688040106"
  uid: 5275f2a8-03fa-4e0f-b8a2-65d1ab80d057
spec:
  deletionPolicy: Delete
  driver: ebs.csi.aws.com
  source:
    volumeHandle: vol-053802929e45c63ed
  sourceVolumeMode: Filesystem
  volumeSnapshotClassName: csi-snapclass
  volumeSnapshotRef:
    apiVersion: snapshot.storage.k8s.io/v1
    kind: VolumeSnapshot
    name: qdrant-storage-qdrant-fe0d11fe-412d-4d35-8c4c-fc694caba63f-0-1725457335
    namespace: qdrant-cloud-client-standard-tier
    resourceVersion: "688039510"
    uid: dfeaf5ce-d193-4ec8-8e87-f5db2cf5464a
status:
  creationTime: 1725457336384000000
  readyToUse: true
  restoreSize: 8589934592
  snapshotHandle: snap-08d3b8119201dede3

```